### ISSUE TITLE
Feature Request - Thermal Runaway Shutdown

### ISSUE THREAD

**Issue by dgarlans** [jneilliii/OctoPrint-WemoSwitch#3]
Is there a way to connect the plugin to Octoprint's thermal panel, so that if the bed temperature goes into runaway somehow, the plugin can trigger a wemo to kill the power to the printer?

My printer failed hot one time and I walked into the room with an 120c bed, and it would be awesome to have a way to prevent that.

**Comment by jneilliii** [comment]
Sorry, but not familiar with the thermal panel you are referring to? As it stands now, no there is no events outside of GCODE commands to trigger an on/off of the wemo device.

**Comment by colemorgan** [comment]
@jneilliii could you monitor the printer temp responses and create a threshold that way?

**Comment by jneilliii** [comment]
Yeah, that would be an option.  I'll see about merging that in on the next updates I do on this plugin.

**Comment by BigWebstas** [comment]
@jneilliii could you merge that into your TPLink plugin as well?

**Comment by jneilliii** [comment]
New release version ready that incorporates thermal runaway protection.  Please test and let me know how it goes.  Install using the following url in plugin manager.

https://github.com/jneilliii/OctoPrint-WemoSwitch/archive/0.1.3.zip

**Comment by jediknight2** [comment]
I just tried to test it and it doesn't seem to be working.  I'm willing to work with you to get it working as I have a 120V bed and the SSR failed the other day. It would be AWESOME for OctoPrint and your Plugin to see the temperature going high and trigger the OFF on the WeMo switch.

WeMo Firmware Version WW_2.00.10626
Hardware Version v1, so this is an older WeMo smart plug.  I just ordered a newer one and will test that one when it comes in

**Comment by jediknight2** [comment]
Here is what I am seeing in log.  I can turn power on and off via the WeMo app
```
[2019-06-10 19:35:14,816] DEBUG: Checking status of 192.168.0.15.
[2019-06-10 19:35:14,817] DEBUG: IP 192.168.0.15 is valid.
[2019-06-10 19:35:14,817] DEBUG: Attempting to connect to 192.168.0.15
[2019-06-10 19:35:24,837] DEBUG: Getting device info from http://192.168.0.15/setup.xml
[2019-06-10 19:35:41,882] DEBUG: Checking status of 192.168.0.15.
[2019-06-10 19:35:41,883] DEBUG: IP 192.168.0.15 is valid.
[2019-06-10 19:35:41,883] DEBUG: Attempting to connect to 192.168.0.15
[2019-06-10 19:35:51,902] DEBUG: Getting device info from http://192.168.0.15/setup.xml
[2019-06-10 19:36:33,770] DEBUG: Max bed temp reached, shutting off plugs.
[2019-06-10 19:36:33,774] DEBUG: Turning off 192.168.0.15.
[2019-06-10 19:36:33,777] DEBUG: {'autoConnectDelay': 10, 'autoConnect': True, 'sysCmdOn': False, 'displayWarning': True, 'ip': '192.168.0.15', 'btnColor': '#808080', 'sysRunCmdOn': '', 'sysRunCmdOff': '', 'autoDisconnect': False, 'label': '3D Printer', 'autoDisconnectDelay': 0, 'sysCmdOff': False, 'thermal_runaway': True, 'sysCmdOffDelay': 0, 'gcodeOffDelay': 1, 'sysCmdOnDelay': 0, 'gcodeEnabled': False, 'gcodeOnDelay': 1, 'currentState': 'unknown', 'warnPrinting': True, 'icon': 'icon-bolt'}
[2019-06-10 19:36:33,778] DEBUG: IP 192.168.0.15 is valid.
[2019-06-10 19:36:33,787] DEBUG: Attempting to connect to 192.168.0.15
[2019-06-10 19:36:35,777] DEBUG: Max bed temp reached, shutting off plugs.
[2019-06-10 19:36:35,780] DEBUG: Turning off 192.168.0.15.
[2019-06-10 19:36:35,783] DEBUG: {'autoConnectDelay': 10, 'autoConnect': True, 'sysCmdOn': False, 'displayWarning': True, 'ip': '192.168.0.15', 'btnColor': '#808080', 'sysRunCmdOn': '', 'sysRunCmdOff': '', 'autoDisconnect': False, 'label': '3D Printer', 'autoDisconnectDelay': 0, 'sysCmdOff': False, 'thermal_runaway': True, 'sysCmdOffDelay': 0, 'gcodeOffDelay': 1, 'sysCmdOnDelay': 0, 'gcodeEnabled': False, 'gcodeOnDelay': 1, 'currentState': 'unknown', 'warnPrinting': True, 'icon': 'icon-bolt'}
[2019-06-10 19:36:35,785] DEBUG: IP 192.168.0.15 is valid.
[2019-06-10 19:36:35,786] DEBUG: Attempting to connect to 192.168.0.15
[2019-06-10 19:36:37,774] DEBUG: Max bed temp reached, shutting off plugs.
[2019-06-10 19:36:37,779] DEBUG: Turning off 192.168.0.15.
[2019-06-10 19:36:37,782] DEBUG: {'autoConnectDelay': 10, 'autoConnect': True, 'sysCmdOn': False, 'displayWarning': True, 'ip': '192.168.0.15', 'btnColor': '#808080', 'sysRunCmdOn': '', 'sysRunCmdOff': '', 'autoDisconnect': False, 'label': '3D Printer', 'autoDisconnectDelay': 0, 'sysCmdOff': False, 'thermal_runaway': True, 'sysCmdOffDelay': 0, 'gcodeOffDelay': 1, 'sysCmdOnDelay': 0, 'gcodeEnabled': False, 'gcodeOnDelay': 1, 'currentState': 'unknown', 'warnPrinting': True, 'icon': 'icon-bolt'}
[2019-06-10 19:36:37,784] DEBUG: IP 192.168.0.15 is valid.
[2019-06-10 19:36:37,785] DEBUG: Attempting to connect to 192.168.0.15
[2019-06-10 19:36:39,780] DEBUG: Max bed temp reached, shutting off plugs.
[2019-06-10 19:36:39,783] DEBUG: Turning off 192.168.0.15.
[2019-06-10 19:36:39,786] DEBUG: {'autoConnectDelay': 10, 'autoConnect': True, 'sysCmdOn': False, 'displayWarning': True, 'ip': '192.168.0.15', 'btnColor': '#808080', 'sysRunCmdOn': '', 'sysRunCmdOff': '', 'autoDisconnect': False, 'label': '3D Printer', 'autoDisconnectDelay': 0, 'sysCmdOff': False, 'thermal_runaway': True, 'sysCmdOffDelay': 0, 'gcodeOffDelay': 1, 'sysCmdOnDelay': 0, 'gcodeEnabled': False, 'gcodeOnDelay': 1, 'currentState': 'unknown', 'warnPrinting': True, 'icon': 'icon-bolt'}
[2019-06-10 19:36:39,787] DEBUG: IP 192.168.0.15 is valid.
[2019-06-10 19:36:39,788] DEBUG: Attempting to connect to 192.168.0.15
[2019-06-10 19:36:41,916] DEBUG: Checking status of 192.168.0.15.
[2019-06-10 19:36:41,917] DEBUG: IP 192.168.0.15 is valid.
```

**Comment by jneilliii** [comment]
@jediknight2 does the plug turn on and off with the button in the navbar?

**Comment by jediknight2** [comment]
OK, just tested and it's working!  I forgot to update my IP when I changed over to the WeMo V2 plug this evening.  Just for clarification for others.  The plugin will NOT work with the older WeMo plug that's labelled "Belkin" and is taller than it is wide.  It DOES work with the new V2 WeMo plug that's labelled "WeMo" and is wider than it is tall.

Thank you for the reply and creating this plugin.  Will make me feel better!  Not sure what's going on with the old WeMo.  The icon always stayed gray and just spun when I tried to cycle power.  The WeMo App DID ask for a firmware upgrade for that plug and when I have time I will upgrade the firmware on the OLD WeMo and try again.  I will report back if anything changed.
